llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1028,6 +1029,14 @@ PassBuilder::buildModuleSimplificationPipeline(OptimizationLevel Level,
       Phase != ThinOrFullLTOPhase::ThinLTOPostLink)
     MPM.addPass(SampleProfileProbePass(TM));
 
+  // Instrument function entry and exit before all inlining.
+  if (Phase != ThinOrFullLTOPhase::ThinLTOPostLink &&
+      Phase != ThinOrFullLTOPhase::FullLTOPostLink &&
+      Phase != ThinOrFullLTOPhase::None) {
```

Comment:
we need this pass when `Phase == ThinOrFullLTOPhase::None` right?

I'd extract out `isLTOPostLink` to mirror the existing `isLTOPreLink`

Reply:
I was confused by setting None prior to that in `buildPerModuleDefaultPipeline`, but it is never called for this assignment to happen. Indeed, since None means non-LTO cases, we should instrument with None. Done.